rae_hw/src/peripherals/mic.cpp

```diff
@@ -9,10 +9,16 @@ namespace rae_hw
 {
     MicNode::MicNode(const rclcpp::NodeOptions &options) : Node("mic_node", options), handle_(nullptr)
     {
-        publisher_ = this->create_publisher<audio_msgs::msg::Audio>("audio_in", 10);
+        publisher_ = this->create_publisher<rae_msgs::msg::RAEAudio>("audio_in", 10);
         configure_microphone();
+
         timer_ = this->create_wall_timer(
             std::chrono::milliseconds(10), std::bind(&MicNode::timer_callback, this));
+        wav_filename_="test.wav";
```

Comment:
On second thought, I think it could be fine if it's set to for example `/tmp/recording.wav`, if the user doesn't specify the path it should be given as a result callback